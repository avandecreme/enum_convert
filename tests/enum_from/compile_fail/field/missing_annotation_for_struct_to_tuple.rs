use enum_convert::EnumFrom;

enum Source {
    Struct { aa: i32, bb: i32 },
}

#[derive(EnumFrom)]
#[enum_from(Source)]
enum Target {
    #[enum_from(Source::Struct)]
    Tuple(
        #[enum_from(Source::Struct.aa)] i32,
        i64, // Annotation #[enum_from(Source::Struct.aa)] is missing
    ),
}

fn main() {}
