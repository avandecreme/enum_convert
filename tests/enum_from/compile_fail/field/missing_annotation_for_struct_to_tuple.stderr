error: Missing required mapping to named field for Source::Struct
  --> tests/enum_from/compile_fail/field/missing_annotation_for_struct_to_tuple.rs:13:9
   |
13 |         i64, // Annotation #[enum_from(Source::Struct.aa)] is missing
   |         ^^^
